tests/actor/test_data_sourcing.py

```diff
@@ -3,92 +3,285 @@
+            Sample[Quantity], active_power_request_6.get_channel_name()
+        ).new_receiver()
+        await req_sender.send(active_power_request_6)
+
+        soc_request_9 = ComponentMetricRequest(
             "test-namespace", 9, ComponentMetricId.SOC, None
         )
-        soc_recv = registry.get_or_create(
-            Sample[Quantity], soc_request.get_channel_name()
+        soc_recv_9 = registry.get_or_create(
+            Sample[Quantity], soc_request_9.get_channel_name()
         ).new_receiver()
-        await req_sender.send(soc_request)
+        await req_sender.send(soc_request_9)
 
-        soc2_request = ComponentMetricRequest(
+        soc2_request_9 = ComponentMetricRequest(
             "test-namespace", 9, ComponentMetricId.SOC, None
         )
-        soc2_recv = registry.get_or_create(
-            Sample[Quantity], soc2_request.get_channel_name()
+        soc2_recv_9 = registry.get_or_create(
+            Sample[Quantity], soc2_request_9.get_channel_name()
         ).new_receiver()
-        await req_sender.send(soc2_request)
+        await req_sender.send(soc2_request_9)
 
-        for _ in range(3):
-            sample = await soc_recv.receive()
+        active_power_request_12 = ComponentMetricRequest(
+            "test-namespace", 12, ComponentMetricId.ACTIVE_POWER, None
+        )
+        active_power_recv_12 = registry.get_or_create(
+            Sample[Quantity], active_power_request_12.get_channel_name()
+        ).new_receiver()
+        await req_sender.send(active_power_request_12)
+
+        for i in range(3):
+            sample = await active_power_recv_4.receive()
             assert sample.value is not None
-            assert 9.0 == sample.value.base_value
+            assert 100.0 + i == sample.value.base_value
 
-            sample = await soc2_recv.receive()
+            sample = await reactive_power_recv_4.receive()
             assert sample.value is not None
-            assert 9.0 == sample.value.base_value
+            assert 100.0 + i == sample.value.base_value
 
-            sample = await active_power_recv.receive()
+            sample = await active_power_recv_6.receive()
             assert sample.value is not None
-            assert 100.0 == sample.value.base_value
+            assert 0.0 + i == sample.value.base_value
 
-        assert await server.graceful_shutdown()
-        connection_manager._CONNECTION_MANAGER = (  # pylint: disable=protected-access
-            None
-        )
+            sample = await soc_recv_9.receive()
+            assert sample.value is not None
+            assert 9.0 + i == sample.value.base_value
+
+            sample = await soc2_recv_9.receive()
+            assert sample.value is not None
+            assert 9.0 + i == sample.value.base_value
+
+            sample = await active_power_recv_12.receive()
+            assert sample.value is not None
+            assert -13.0 + i == sample.value.base_value
```

Comment:
I think it is not a good idea to use the `==` operator to compare a floating-point value against the result of an addition operation involving floating-point numbers, even in tests. I'd suggest to use isclose() from pytest or math instead

Reply:
Can you please create a separate issue? I'm just changing the indentation to this code.

Reply:
ok, will create an issue

Reply:
https://github.com/frequenz-floss/frequenz-sdk-python/issues/965